kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/VaultTemplateInvocationContext.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.encryption;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.support.TypeBasedParameterResolver;
+import org.testcontainers.vault.VaultContainer;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.filter.encryption.EnvelopeEncryption;
+import io.kroxylicious.filter.encryption.TemplateKekSelector;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultKms;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultKmsService;
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultResponse;
+import io.kroxylicious.proxy.config.FilterDefinition;
+import io.kroxylicious.proxy.config.FilterDefinitionBuilder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VaultTemplateInvocationContext implements TestTemplateInvocationContext {
+    private static final String VAULT_TOKEN = "rootToken";
+    private static final String HASHICORP_VAULT = "hashicorp/vault:1.15";
```

Comment:
does this need to be kept in sync with the version defined in the pom?

Reply:
There is a default in org.testcontainers.vault.VaultContainer#DEFAULT_TAG but unfortunately it is really old (1.1.3) so I think we need our own constant.  I do want to the constant to come from a single location.